### PR TITLE
[#51745] Consolidate translations strings for work package sharing feature

### PR DESCRIPTION
**Pull request by dombesz**
https://community.openproject.org/work_packages/51745

### CODE REVIEW

File: config/locales/js-en.yml [hunk 1]
```diff
@@ -1123,7 +1123,7 @@ en:
       sharing:
         share: "Share"
         title: "Share work package"
-        show_all_users: "Show all shared users"
+        show_all_users: "Show all users with whom the work packages has been shared with"
```
Comment (dombesz) [inline config/locales/js-en.yml]: According to the docs, the meaning of the translation was incorrect. Here's my proposed version.
cc: @psatyal

File: config/locales/en.yml [hunk 2]
```diff
@@ -357,6 +357,7 @@ Project attributes and sections are defined in the <a href=%{admin_settings_url}
       project_roles: "Project roles"
       wp_shares: "Work package shares"
       groups: "Groups"
+    no_modify_on_shared: "You currently cannot modify or remove shared memberships through the member page. Use the sharing modal instead."
```
Comment (dombesz) [inline config/locales/en.yml:360]: This translation was completely missing and it is not being used. Added it for future use.

**Review by cbliard — APPROVED** [review]
LGTM.
I only fixed one typo in `query_fields.shared_with_user`.

**Comment by dombesz** [comment]
Thanks @cbliard, I am not sure if that is a typo. According to the specification it should be "users".

image

**Comment by cbliard** [comment]
@dombesz well, I called it a typo because I assumed that the person having added that string made a typo when entering it.
I [fixed it](https://github.com/opf/openproject/pull/15533/commits/925f69797afe7e735e8a2e4d25fc60b41474e088) to match the specification so it's "users" as expected.

**Comment by dombesz** [comment]
@cbliard you are right, I overlooked that one, thanks!

**Pull request merged**